### PR TITLE
Update to pkgdown and re-exporting some gtsummary functions

### LINKED ISSUE #25: Customize `_pkgdown.yml`

### LINKED ISSUE #30: How to make gtsummary helper functions available?
There are a bunch of gtsummary functions that will be helpful for formatting and styling gtreg tables. How should we make them available to the users, e.g. the `modify_*()` family, bolding/italicizing, themes.

1. Re-export the functions from gtsummary to gtreg? A downside to this approach is that any new functions will need to be explicitly added to the re-export list.
2. Load gtsummary with gtreg? This would work, but I typically don't like it when a package loads another with it. It would also attach a bunch of functions the users don't need, e.g. `tbl_cross()`, `tbl_svysummary()`. Would make the autocomplete somewhat less helpful.
3. We could also just not export any of them, and users can load gtsummary if needed.

### PR DESCRIPTION
**Pull request by ddsjoberg**
- The beginnings of a pkgdown site have been added (this can of course be updated)
- Re-exporting gtsummary functions

closes #30 
closes #25

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,18 +1,21 @@
 Package: gtreg
 Title: What the Package Does (One Line, Title Case)
 Version: 0.0.0.9000
-Authors@R: 
-    person("First", "Last", , "first.last@example.com", role = c("aut", "cre"),
-           comment = c(ORCID = "YOUR-ORCID-ID"))
+Authors@R: c(
+    person("Shannon", "Pileggi", , "first.last@example.com", role = c("aut", "cre"),
+           comment = c(ORCID = "YOUR-ORCID-ID")),
```
Comment (ddsjoberg) [inline DESCRIPTION:6]: @shannonpileggi can you add your ORCID here, or delete the comment?

Comment (shannonpileggi) [inline DESCRIPTION:6]:
```suggestion
person("Shannon", "Pileggi", , "shannon@pipinghotdata.com", role = c("aut", "cre"),
           comment = c(ORCID = "0000-0002-7732-4164")),
```

File: man/gtreg-package.Rd [hunk 2]
```diff
@@ -16,7 +16,12 @@ Useful links:
 
 }
 \author{
-\strong{Maintainer}: First Last \email{first.last@example.com} (\href{https://orcid.org/YOUR-ORCID-ID}{ORCID})
+\strong{Maintainer}: Shannon Pileggi \email{first.last@example.com} (\href{https://orcid.org/YOUR-ORCID-ID}{ORCID})
```
Comment (shannonpileggi) [inline man/gtreg-package.Rd:19]: This is auto generated from `DESCRIPTION` right? So I don't need to edit this?

File: DESCRIPTION [hunk 3]
```diff
@@ -1,18 +1,21 @@
 Package: gtreg
 Title: What the Package Does (One Line, Title Case)
```
Comment (shannonpileggi) [inline DESCRIPTION:2]:
```suggestion
GT Backed Regulatory Tables For Clinical Research
```

A working title?

File: DESCRIPTION (hunk 1 above)
Comment (shannonpileggi) [reply DESCRIPTION:6]: Wasn't sure if I had one, but found it!

**Review by shannonpileggi — CHANGES_REQUESTED** [review]
Thanks for keeping this moving!

File: man/gtreg-package.Rd (hunk 2 above)
Comment (ddsjoberg) [reply man/gtreg-package.Rd:19]: right!

File: DESCRIPTION (hunk 3 above)
Comment (ddsjoberg) [reply DESCRIPTION:2]: We should look up the CRAN rules...i think they have a policy on how to refer to another package's name in a title or description.

File: DESCRIPTION [hunk 4]
```diff
@@ -1,18 +1,21 @@
 Package: gtreg
-Title: What the Package Does (One Line, Title Case)
+Title: GT Backed Regulatory Tables For Clinical Research
```
Comment (ddsjoberg) [inline DESCRIPTION:2]: Found it! "The mandatory ‘Title’ field should give a short description of the package. Some package listings may truncate the title to 65 characters. It should use title case (that is, use capitals for the principal words: tools::toTitleCase can help you with this), not use any markup, not have any continuation lines, and not end in a period (unless part of …). Do not repeat the package name: it is often used prefixed by the name. Refer to other packages and external software in single quotes, and to book titles (and similar) in double quotes."

I'll update it

**Review by shannonpileggi — APPROVED** [review]
Thanks for the title adjustment, this looks great!

**Pull request merged**